Add alpha 7 changelog.

vpetrovykh opened the pull request.

1st1 commented on docs/changelog/1_0_a7.rst:
```diff
@@ -0,0 +1,86 @@
+EdgeQL
+======
+
+* Implement consistent update of expressions when referenced schema
+  objects are renamed (:eql:gh:`#1841`).
```
Overall this looks like you simply copy/pasted git log here. We need more editing, grouping of related commit messages, and some examples.

1st1 commented on docs/changelog/1_0_a7.rst:
```diff
@@ -0,0 +1,86 @@
+  driver forward compatibility with the new edgedb protocol versions.
+* Update the `edgedb-python <https://github.com/edgedb/edgedb-python>`_ driver
+  to 0.12.0a1.
+* Allow configuring __tid__ injected property via private APIs in
```
this shouldn't be mentioned

vpetrovykh replied:
OK

1st1 commented on docs/changelog/1_0_a7.rst:
```diff
@@ -0,0 +1,86 @@
+* Move request methods into Executor interface (`#76
+  <https://github.com/edgedb/edgedb-js/pull/76>`_) as part of the `RFC
+  1004 <robust_>`_ changes.
+* Fix `edgedb-python <https://github.com/edgedb/edgedb-python>`_
```
remove this

vpetrovykh replied:
I thought that forward compatibility was a noteworthy enough feature of any tool that interacts with other tools.

1st1 replied:
This is just a minor bug fix. We typically don't mention these in changelogs, and neither do other projects.

vpetrovykh replied:
Removed

1st1 commented on docs/changelog/1_0_a7.rst:
```diff
@@ -0,0 +1,86 @@
+  1004 <robust_>`_ changes.
+* Fix `edgedb-python <https://github.com/edgedb/edgedb-python>`_
+  driver forward compatibility with the new edgedb protocol versions.
+* Update the `edgedb-python <https://github.com/edgedb/edgedb-python>`_ driver
```
remove this

vpetrovykh replied:
We normally reported driver version changes, why not now?

1st1 replied:
Right, let's keep this, but change the version to 0.12.0.  We should never mention alpha versions in our chnagelogs.

vpetrovykh replied:
Removed

1st1 commented on docs/changelog/1_0_a7.rst:
```diff
@@ -0,0 +1,86 @@
+  to 0.12.0a1.
+* Allow configuring __tid__ injected property via private APIs in
+  python driver (`#105 <https://github.com/edgedb/edgedb-python/pull/105>`_).
+* Allow passing ``implicit_typenames`` header in the python driver (`#105
```
remove this

vpetrovykh replied:
OK

1st1 commented on docs/changelog/1_0_a7.rst:
```diff
@@ -0,0 +1,86 @@
+Rolling out the support for the full spec will be done in the next
+release, but some implementation work has already started.
+
+* Initial implementation of capabilities from `RFC 1004 <robust_>`_
```
remove this

vpetrovykh replied:
I was thinking that much like we had an ongoing effort to implement RFC1000 mentioned in past changelogs (alpha 4, 5, 6), a similar mention is worth for starting implementaiton of RFC1004.

1st1 replied:
This has no effect on user APIs -- it's an internal change.

vpetrovykh replied:
Removed

1st1 commented on docs/changelog/1_0_a7.rst:
```diff
@@ -0,0 +1,86 @@
+  fixing issues with changing bases, improving expression dependency
+  tracing, improving handling changes to aliases, adding DDL commands
+  to better control fields like ``errmessage`` or ``default`` and
+  other changes (:eql:gh:`#1987`).
```
Most of items here are individual commit message. Most are related to migrations. I suggest grouping all migration related changes in its own section and elaborating on them.

vpetrovykh replied:
OK, I'll make a separate section for migrations because it is the motivation behind a lot of the individual changes. Pure EdgeQL changes will shrink to things like `DATABASE` config and `bigint` vs. `decimal` clarification.

I'll add some kind of introductory paragraph to the migration section.

1st1 commented on docs/changelog/1_0_a7.rst:
```diff
@@ -0,0 +1,86 @@
+  python driver (`#105 <https://github.com/edgedb/edgedb-python/pull/105>`_).
+* Allow passing ``implicit_typenames`` header in the python driver (`#105
+  <https://github.com/edgedb/edgedb-python/pull/105>`_).
+* Fix the python driver query cache to account for expected
```
remove this

vpetrovykh replied:
OK

1st1 reviewed (commented):
This needs more work.

vpetrovykh commented:
I've added a few examples for some of the features and moved all migration stuff into its own section. Now migration also has an example + list of features that got improvements, so unlike in previous releases this one will look like the migrations are a lot more fleshed out.

I'm not sure about why should some of the items you marked for removal should be removed (I've left comments why I included them), but feel free to drop them of course.

I've also made a few tiny tweaks to the actual docs.

vpetrovykh commented:
Also, I ran into a tokenizer bug while I was at this (thanks to the doc snippets tests) - https://github.com/edgedb/edgedb/pull/2040

For the moment I just avoid the offending code, but that's weird how this remained unnoticed for so long. I think the constant extractor is typically masking this.

tailhook commented on docs/changelog/1_0_a7.rst:
```diff
@@ -0,0 +1,218 @@
+==================
+
+* Deprecate ``--default-database`` and ``--default-database-user``
+  (:eql:gh:`#1879`).
```
This is not `Command-line Tools` change, it's server command-line.

vpetrovykh replied:
OK

1st1 replied:
@vpetrovykh Have you added what Paul suggested yet?

vpetrovykh replied:
Yes

tailhook commented on docs/changelog/1_0_a7.rst:
```diff
@@ -0,0 +1,218 @@
+
+* Deprecate ``--default-database`` and ``--default-database-user``
+  (:eql:gh:`#1879`).
+* Add a new header to the ``Prepare`` message to implicitly inline
```
Don't think we need this

vpetrovykh replied:
Removed

tailhook reviewed (commented):
On the actual command-line tools:
1. Add ``edgedb server logs``
2. Add ``--connect-timeout``
3. Add ``--dsn``
4. Bugfix: `\lt` doesn't show cryptic internal types (unions and intersections)
5. Bugfix: added shutdown timeout of second (previously could hang indefinitely)
6. Default user is now `edgedb` and default database `edgedb`. Default database is not named as a user any more.
7. When installing server DEBIAN_FRONTEND is now `noninteractive` by default and is overridable
8. Bugfix: repl now supports full range of datetime values
9. Bugfix: ``server install`` now friendlier on linuxes without systemd allowing foreground run
10. Repl by default shows typenames, and effectively `\set introspect-types` setting is removed
11. Non-interactive error messages are prefixed by `edgedb error`, to quickly spot which tool has errored in scripts
12. Add `migration-log` command
13. Add `server info` command *(should merge today)*

1st1 reviewed (approved):
Much better!

1st1 commented:
@elprans want to take a look before we merge?

vpetrovykh commented:
One sec! I'm adding the CLI update from @tailhook

elprans commented on docs/changelog/1_0_a7.rst:
```diff
@@ -0,0 +1,242 @@
+.. code-block:: edgeql-repl
+
+    db> \set output-mode json
+    db[tx]> DESCRIBE CURRENT MIGRATION AS JSON;
```
Why are we exhibiting raw `DESCRIBE` instead of the `create-migration` CLI?

vpetrovykh replied:
Largely because I'm not as familiar with create-migration CLI (since I pretty much use `DESCRIBE` in tests), but also because it is smaller in terms of explanations needed for the example, like you don't need to touch on the schema directory and file organization in it to do an example. The full-blown `create-migration` CLI example might be good fodder for the blog? Either for the release or even a standalone one?

I guess this is a work-in-progress vs. finished tool consideration. For work-in-progress showing lower level guts can be more interesting because it shows capability (like JSON that can be integrated with a tool), whereas when the tool is finished you probably want to show off how nice and smooth it is without bothering with the internals. So although it's my personal experience bias that lead me to do `DESCRIBE` in REPL example, but I think it's a good insight into our work at this stage.

elprans replied:
OK, fair enough.

> but I think it's a good insight into our work at this stage.

Please make this clear in the description.  `DESCRIBE CURRENT MIGRATION AS JSON` is not intended to be used directly.

vpetrovykh replied:
I've added another paragraph mentioning the tool and that the example is showing low-level commands.

tailhook commented on docs/changelog/1_0_a7.rst:
```diff
@@ -0,0 +1,248 @@
+* When installing server ``DEBIAN_FRONTEND`` is now ``noninteractive`` by
+  default and is overridable (`#188
+  <https://github.com/edgedb/edgedb-cli/pull/188>`_).
+* Add ``edgedb server logs`` (`#172
```
nit: either use `edgedb server xxx` everywhere or `server xxx` everywhere (I prefer former), currently it's inconsinstent.

vpetrovykh replied:
Updated

tailhook commented on docs/changelog/1_0_a7.rst:
```diff
@@ -0,0 +1,248 @@
+Server Command-Line
+===================
+
+* Make ``server install`` friendlier on linuxes without systemd
```
I'm still not sure this fits `Server Command-Line`. I would say only `--default-database` and `--default-database-user` do.

vpetrovykh replied:
I'm kinda going with the idea that tools/commands that interact with the server with the purpose of monitoring/installing/managing some aspect of the server are "server" tools. Whereas the tools focusing on data (so schema or querying) are "regular" CLI.

tailhook commented on docs/changelog/1_0_a7.rst:
```diff
@@ -0,0 +1,248 @@
+* Add ``edgedb server logs`` (`#172
+  <https://github.com/edgedb/edgedb-cli/pull/172>`_).
+* Add ``server info`` command.
+* Add shutdown timeout of 1 second.
```
This is super-unclear. Better:
* When exiting from the REPL with Ctrl+D client now waits max of 1 second for server response

But generally, if you skipped all bugfixes anyway, this might go unnoticed too.

vpetrovykh replied:
Perhaps I could just drop it from the highlights.
